src/Mvc/src/Microsoft.AspNetCore.Mvc.NewtonsoftJson/DependencyInjection/MvcJsonMvcCoreBuilderExtensions.cs

```diff
@@ -80,6 +84,30 @@ internal static void AddJsonFormatterServices(IServiceCollection services)
             services.TryAddEnumerable(
                 ServiceDescriptor.Transient<IApiDescriptionProvider, JsonPatchOperationsArrayProvider>());
             services.TryAddSingleton<JsonResultExecutor>();
+
+            var tempDataSerializer = services.FirstOrDefault(f => f.ServiceType == typeof(TempDataSerializer));
+            if (tempDataSerializer != null)
```

Comment:
Should we look for the exact registration we create by default? We don't want to stomp on things the user created.